editor/src/core/model/element-metadata.spec.browser2.tsx

```diff
@@ -859,6 +867,90 @@ describe('isAutofocusable/isManuallyFocusableComponent', () => {
       'story/scene/app:app-inner-div/inner-app-2: true',
     ])
   })
+
+  it('supports components imported with a file mapped import statement', async () => {
+    const sampleProjectContents = contentsToTree({
+      'jsconfig.json': codeFile(
+        `
+        {
+          "compilerOptions": {
+            "paths": {
+              "~/*": [
+                "app/*"
+              ]
+            }
+          }
+        }
+        `,
+        null,
+      ),
+      '/app/card.js': codeFile(
+        `
+        import * as React from 'react'
+
+        export const Card = (props) => {
+          return (
+            <div style={props.style} data-uid='card-root'>
+              <div
+                style={{
+                  position: 'absolute',
+                  left: 10,
+                  top: 10,
+                }}
+                data-uid='card-root-child'
+              >
+                I am a card
+              </div>
+            </div>
+          )
+        }
+        `,
+        null,
+      ),
+      '/utopia/storyboard.js': codeFile(
+        `
+        import * as React from 'react'
+        import { Storyboard } from 'utopia-api'
+        import { Card } from '~/card'
+
+        export var storyboard = (
+          <Storyboard data-uid='sb'>
+            <Card
+              style={{
+                backgroundColor: '#aaaaaa33',
+                position: 'absolute',
+                top: 0,
+                left: 0,
+                width: 150,
+                height: 150,
+              }}
+              data-uid='card'
+            />
+          </Storyboard>
+        )
+        `,
+        null,
+      ),
+    })
+    const renderResult = await renderTestEditorWithProjectContent(
+      sampleProjectContents,
+      'await-first-dom-report',
+    )
+    const metadata = renderResult.getEditorState().editor.jsxMetadata
+    const pathTrees = renderResult.getEditorState().editor.elementPathTree
+    const autoFocusedPaths = renderResult.getEditorState().derived.autoFocusedPaths
+    const filePathMappings = renderResult.getEditorState().derived.filePathMappings
+    const targetPath = EP.fromString('sb/card')
+    expect(MetadataUtils.isAutofocusable(metadata, pathTrees, targetPath)).toBeFalsy()
+    expect(
+      MetadataUtils.isManuallyFocusableComponent(
+        targetPath,
+        metadata,
+        autoFocusedPaths,
+        filePathMappings,
+      ),
+    ).toBeTruthy()
```

Comment:
Or whatever similar incantation, so that if this ends up returning an array later on this check doesn't keep giving it the thumbs up:
```suggestion
    ).toEqual(true)
```

Reply:
Good shout 👍